### PR TITLE
[orchestration] Update agent-payload from v4.80.0 to v4.82.0 and extract priority class from pod

### PR DESCRIPTION
**Pull request by kangyili**
### What does this PR do?

Update agent-payload from v4.80.0 to v4.82.0
### Motivation

Based on DataDog/datadog-agent#9172 and DataDog/agent-payload#122

### Additional Notes

Anything else we should know when reviewing?

### Describe how to test your changes

Write here in detail how you have tested your changes
and instructions on how this should be tested in QA.

Describe or link instructions to set up environment 
for testing, if the process requires more than just
running the agent on one of the supported platforms.

### Checklist

- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] The `need-change/operator` and `need-change/helm` labels has been applied if applicable.
- [x] The appropriate `team/..` label has been applied, if known.
- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated if applicable.

Note: Adding GitHub labels is only possible for contributors with write access.

### CODE REVIEW

File: pkg/util/orchestrator/pod.go [hunk 1]
```diff
@@ -151,6 +155,8 @@ func extractPodMessage(p *v1.Pod) *model.Pod {
 	podModel.IP = p.Status.PodIP
 	podModel.RestartCount = 0
 	podModel.QOSClass = string(p.Status.QOSClass)
+	// pod priority class
```
Comment (nammn) [inline pkg/util/orchestrator/pod.go:158]: nit: we can remove this comment, that doesn't give us any additional info

File: pkg/util/orchestrator/pod.go [hunk 2]
```diff
@@ -67,6 +68,9 @@ func ProcessPodList(podList []*v1.Pod, groupID int32, hostName string, clusterID
 		// additional tags
 		podModel.Tags = append(tags, fmt.Sprintf("pod_status:%s", strings.ToLower(podModel.Status)))
 
+		// collect priorityClassName
+		podModel.Tags = append(tags, fmt.Sprintf("priority_class:%s", strings.ToLower(podModel.PriorityClass)))
```
Comment (nammn) [inline pkg/util/orchestrator/pod.go:72]: This information will already be there given by this PR https://github.com/DataDog/datadog-agent/pull/9172 in the tagger, therefore you don't need to add this tag here. I suggest to do this in our backend for `datadog-cluster-agent` payloads

File: pkg/util/orchestrator/pod.go [hunk 3]
```diff
@@ -3,6 +3,7 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2016-present Datadog, Inc.
 
+//go:build orchestrator
```
Comment (nammn) [inline pkg/util/orchestrator/pod.go:6]: why this line?

File: pkg/util/orchestrator/pod.go (hunk 2 above)
Comment (nammn) [reply pkg/util/orchestrator/pod.go:72]: + The name doesn't seem to match i.e. from above PR: `kube_priority_class`

**Review by nammn — COMMENTED** [review]
💯  Looks good to me mostly.

My suggestion:

Add a unit test

File: pkg/util/orchestrator/pod.go [hunk 4]
```diff
@@ -151,6 +151,7 @@ func extractPodMessage(p *v1.Pod) *model.Pod {
 	podModel.IP = p.Status.PodIP
 	podModel.RestartCount = 0
 	podModel.QOSClass = string(p.Status.QOSClass)
+	podModel.PriorityClass = string(p.Spec.PriorityClassName)
```
Comment (mfpierre) [inline pkg/util/orchestrator/pod.go:154]: seems that`PriorityClassName` https://github.com/kubernetes/api/blob/96087243fecba390391364dd42abe2f3ff21c1e3/core/v1/types.go#L3124 is already a string so the extra cast is not necessary

Comment (kangyili) [reply pkg/util/orchestrator/pod.go]: Done

**Comment by nammn** [comment]
@kangyili if you didn't see it yet, please follow this guideline: https://github.com/DataDog/datadog-agent/blob/0311432e7b2dc94182a71b753b965c6e5c346f4c/docs/dev/contributing.md#pull-requests (hit me up if some things are unclear)

File: releasenotes/notes/Add-priority-class-0e935f5af83abd51.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# Each section from every release note are combined when the
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+upgrade:
```
Comment (nammn) [inline releasenotes/notes/Add-priority-class-0e935f5af83abd51.yaml:9]: I think "features" is historically the better section 👍

File: releasenotes/notes/Add-priority-class-0e935f5af83abd51.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+  - |
+    [Orchestrator]Extract priority class from pod and update unit test
```
Comment (nammn) [inline releasenotes/notes/Add-priority-class-0e935f5af83abd51.yaml:11]:
```suggestion
      priority class name of pods are now collected and sent to the orchestration endpoint
```

**Review by nammn — APPROVED** [review]
Great work 💯

**Pull request merged**